spec/result.json

```diff
@@ -0,0 +1,73 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "required": [
+        "head"
+    ],
+    "properties": {
+        "head": {
+            "type": "object",
+            "properties": {
+                "additionalProperties": false,
+                "link": {
+                    "type": "array",
+                    "items": {
+                        "type": "string",
+                        "format": "uri"
+                    }
+                },
+                "vars": {
+                    "type": "array",
+                    "items": {
+                        "type": "string"
+                    }
+                }
+            }
+        },
+        "boolean": {
+            "type": "boolean"
+        },
+        "results": {
+            "type": "object",
+            "required": [
+                "bindings"
+            ],
+            "properties": {
+                "bindings": {
+                    "type": "array",
+                    "items": {
+                        "type": "object",
+                        "additionalProperties": {
+                            "type": "object",
+                            "properties": {
+                                "type": {
+                                    "type": "string",
+                                    "enum": [
+                                        "uri",
+                                        "literal",
+                                        "bnode"
+                                    ]
+                                },
+                                "value": {
+                                    "type": "string"
+                                },
+                                "xml:lang": {
+                                    "type": "string",
+                                    "maxLength": 35
```

Comment:
Idea: use the same regular expression as Turtle.

Reply:
Could do, although the results capturing general information from the RDF abstract data model, which refers to BCP47.
The XML results schema does not have such restrictions.

The full regular expression is ghastly!
https://stackoverflow.com/questions/7035825/regular-expression-for-a-language-tag-as-defined-by-bcp47

The JSON schema regex definition is [specific subset of Javascript](https://json-schema.org/understanding-json-schema/reference/regular_expressions.html) and not language independent.

The value of having a schema is having a formal, executable description of the structure of the SPARQL JSON results described [section 4](https://w3c.github.io/sparql-results-json/spec/index.html#select-results-form) and [section 5](https://w3c.github.io/sparql-results-json/spec/index.html#ask-result-form).

Let's focus on structure which is recording what is already in the results document.

Reply:
Yes, 35 chars are the maximum of a valid language tag in the context of  BCP47. It is possible to use regex, but it will be strongly complicated (and I think that won't be supported by JSON Schema validators). So I'd rather avoid regexs. Another option is to remove it.